### PR TITLE
Migrate from dep to new go modules support in go 1.11.x

### PR DESCRIPTION
**Pull request by mestery**
Some documentation on go modules is found here [1].

The following is done:

- executed `go mod init` to migrate from dep's Gopkg.[lock, toml]
- removed the vendor directory except for the code-generators which
  are required for the build
- removed dep management from the Makefile
- removed dep cehck from travis

This is similar to the work done on this [1] vpp-agent PR.

[1] https://github.com/golang/go/wiki/Modules
[2] https://github.com/ligato/vpp-agent/pull/840

Signed-off-by: Kyle Mestery <mestery@mestery.com>

### CODE REVIEW

**Comment by mestery** [comment]
Thanks for the tip on how to do this @ondrej-fabry, I've converted crd-example over to this now. It works quite slick!

**Comment by ondrej-fabry** [comment]
I'd not suggest getting rid of support for go 1.10 just yet. Also note that getting rid of vendor makes Travis builds longer and there is rare possibility of some dependency to not be available at certain point in time. 

However, there is a way to satisfy both, at least for the time being until the situation around modules settles a little more, perhaps go 1.12. I recommend you to read this article that nicely explains how to do this: https://arslan.io/2018/08/26/using-go-modules-with-vendor-support-on-travis-ci/

**Comment by mestery** [comment]
@ondrej-fabry That's a good pointer, thanks! Note that `go mod vendor` removes the code-generators, which is what I'm trying to sort out now. We have no direct dependency on those in the code, but we call them to update CRD objects. Trying to figure out how to get those to remain and I think I can make this work with go v1.10.x and go v1.11.x.

**Comment by ondrej-fabry** [comment]
It seems recommended approach to vendor tools is to create `tools.go` file with `// +build tools` build constraint and list them as blank imports. This will force `go mod vendor` to vendor even those dependencies, but exclude from normal build.

https://github.com/golang/go/wiki/Modules#how-can-i-track-tool-dependencies-for-a-module

File: scripts/build.sh [hunk 1]
```diff
@@ -19,8 +19,14 @@ while test $# -gt 0; do
 done
 
 [ -d vendor/github.com/ligato/crd-example/ ] && (echo "Run: rm -rf vendor/github.com/ligato/crd-example;dep ensure";exit 1)
-test -z "$(go fmt ./...)" || (echo "Run go fmt ./... and recommit your code";exit 1)
-go get -u github.com/golang/protobuf/protoc-gen-go
-go generate ./...
-go install ./...
-go test $RACE_ENABLED ./...
+test -z "$(env GO111MODULE=on go fmt ./...)" || (echo "Run go fmt ./... and recommit your code";exit 1)
+env GO111MODULE=on go get -u github.com/golang/protobuf/protoc-gen-go
```
Comment (ondrej-fabry) [inline scripts/build.sh]: Perhaps, you could just add `export GO111MODULE=on` to the beginning of the file.

**Pull request merged**